### PR TITLE
chore(argo-events): use policy/v1 for poddisruptionbudgets, when available

### LINKED ISSUE #1709: [argo-events] PodDisruptionBudget policy/v1beta1 deprecated from k8s 1.22+
### Describe the bug

from https://kubernetes.io/docs/reference/using-api/deprecation-guide/#poddisruptionbudget-v125:
`The policy/v1beta1 API version of PodDisruptionBudget will no longer be served in v1.25.
Migrate manifests and API clients to use the policy/v1 API version, available since v1.21.`

### Related helm chart

argo-events

### Helm chart version

2.0.7

### To Reproduce

files in chart that still use `policy/v1beta1`

### Expected behavior

something like this
```
{{ if $.Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" -}}
apiVersion: policy/v1
{{- else -}}
apiVersion: policy/v1beta1
{{- end }}
```

should work.

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by brainiac84**
Signed-off-by: Vadim Grek <vadimprog@gmail.com>

Fixes #1709

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Comment by brainiac84** [comment]
@jbehling @Vaibhav or @pdrastil mind to check

File: charts/argo-events/Chart.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ maintainers:
     url: https://argoproj.github.io/
 annotations:
   artifacthub.io/changes: |
+    - "[Fixed]: use policy/v1 for poddisruptionbudgets, when available
     - "[Fixed]: avoid app.kubernetes.io/version kubernetes label from exceeding maximum length (63)
```
Comment (pdrastil) [inline charts/argo-events/Chart.yaml:19]: Please remove unrelated fixes from the annotation.

**Comment by pdrastil** [comment]
Hi @brainiac84 thanks for the PR. LGTM, just remove the fixes from previous version in Chart.yaml so they are not published with this change. Thank you

File: charts/argo-events/Chart.yaml (hunk 1 above)
Comment (brainiac84) [reply charts/argo-events/Chart.yaml]: Fixed

**Comment by brainiac84** [comment]
Hi @pdrastil removed unrelated fixes from the annotation.

**Review by pdrastil — APPROVED** [review]
LGTM
